### ISSUE TITLE
Multip

### ISSUE THREAD

**Issue by lousybyte** [DuendeSoftware/Support#1062]
**Which version of Duende IdentityServer are you using?**
6.2

**Which version of .NET are you using?**
.NET 6

**Describe the bug**

We have a special setup that allows our IdP to be accessible via 2 different URLs, and that implies that the issuer value inside the token will be different depending on what URL was used to request it.
For our own APIs, we made sure to add both issuer values to the list of allowed issuers, e.g.:

```
var issuers = new List<string>()
    {
       "https://url_one.org",
      "https://url_two.org"
    };
// ...
TokenValidationParameters tokenValidationParameters = new TokenValidationParameters
{
    // ...

    ValidateIssuer = true,
    ValidIssuers = issuers

    // ...
};
```

The issue we have is, that some applications might request a token via `https://url_one.org` and then try to use the `/connect/userinfo` endpoint for example via `https://url_two.org`, in which case IdentityServer is rejecting the token because of issuer mismatch.

Is there a way to provide a list of issuers that are allowed for the IdentityServer endpoints, the same as we can do for our own APIs in .NET?

**Comment by AndersAbel** [comment]
IdentityServer supports responding on multiple addresses which (unless configured otherwise) creates tokens with different issuers. This is usually done to support some kind of separation between different groups of users, for example multiple tenants in a multi tenanted SAAS solution. The different host names could also be used to direct requests to the user info endpoint to the right user store.

From your description, it sounds like this is not the case you have as you want the tokens to be usable across the host names. Before going on to answer your question I'd like to ask: Do you really need the multiple issuers? You can still respond to different host names but have a fixed configured issuer to make all tokens created have the same issuer.

If you still need to get user info endpoint to accept tokens with multiple issuers that would require replacing/customizing the `ITokenValidator` service. That service is however not designed to be replaced. The implementation is in an internal class with non-virtual and private methods. So while it is possible, it would require quite a lot of work and is not what we would recommend doing.

**Comment by lousybyte** [comment]
Yes, that is correct, the token client and the destination API are in different network configuration zones, e.g. the client is public and the API is private, and while they can communicate between each other, they cannot access the IdP through the same node due to the network configuration as well.

Currently we did set up the issuer as static and that solved the problem, was wondering if maybe we missed something simple in the documentation that allowed multiple issuers during the validation.

Can be closed, thank you for your answer!
